### PR TITLE
[api][webui] Fix attrib policy scope

### LINKED ISSUE #2782: Failed to create the attribute with error 403
Issue/Feature description
====================
As part of factory-staging group, I've the rights in openSUSE:Factory:Staging project, in the past, I've no problem creating a attribute, however I got error 403 now, which affects executing the factory staging tool. I tried api call too, the complete error message below,

<pre>
mlin@rana:~> osc -d api -X POST /source/openSUSE:Factory:Staging/_attribute/openSUSE:LockedBy -T fake_lock
POST https://api.opensuse.org/source/openSUSE:Factory:Staging/_attribute/openSUSE:LockedBy
Server returned an error: HTTP Error 403: Forbidden
Date: Thu, 09 Mar 2017 07:57:40 GMT
Server: Apache/2.4.23 (Linux/SUSE)
Strict-Transport-Security: max-age=31536000
Cache-Control: no-cache
Vary: Accept-Encoding
X-XSS-Protection: 1; mode=block
X-Opensuse-Runtimes: {"view":1.0777139686979353,"db":44.57578399999999,"backend":0,"xml":0.081272}
X-Request-Id: 3de017b5-488d-485a-a977-56631b6c1504
X-Opensuse-APIVersion: 2.7.51.git20170308.8cb06cb
X-Runtime: 0.087399
X-Frame-Options: SAMEORIGIN
X-Content-Type-Options: nosniff
X-Opensuse-Errorcode: not_authorized
X-Powered-By: Phusion Passenger 5.0.30
Status: 403 Forbidden
Content-Type: application/xml; charset=utf-8
Connection: close
Transfer-Encoding: chunked
 
<status code="not_authorized">
  <summary>You are not authorized to create this Attrib.</summary>
</status>
 
You are not authorized to create this Attrib.
</pre>

Expected result
============
No 403 error if I've the proper rights.

How to Reproduce
===============
Doing a api call to create a attribute.

### PR DESCRIPTION
**Pull request by hennevogel**
Only check AttribType rules and not AttribNamespace rules for
the Attrib policy. Fixes #2782 

This required to revert some of the changes @bgeuken did to adop the test suite to this...

### CODE REVIEW

**Comment by hennevogel** [comment]
This is also monkey patched (in a different way) on production

**Comment by evanrolfe** [comment]
LGTM but it would be nice if somebody more familiar with the API could verify that the behaviour described in the tests is correct since I'm not too familiar with how this is meant to work. @adrianschroeter perhaps?

File: src/api/app/policies/attrib_policy.rb [hunk 1]
```diff
@@ -3,13 +3,16 @@ def create?
     # Admins can write everything
     return true if @user.is_admin?
 
-    at_modifiables = modifiables
-    if at_modifiables.empty? && @record.container.present?
+    if @record.attrib_type.nil? || @record.attrib_type.attrib_type_modifiable_bies.empty?
```
Comment (mschnitzer) [inline src/api/app/policies/attrib_policy.rb]: I would change `@record.attrib_type.nil?` to `!@record.attrib_type`

File: src/api/test/functional/issue_controller_test.rb [hunk 2]
```diff
@@ -138,17 +138,6 @@ def test_get_issue_for_linked_packages
...
-    data = "<attributes><attribute namespace='OBS' name='Issues'>
-              <issue name='987' tracker='bnc'/>
-              <issue name='654' tracker='bnc'/>
-            </attribute></attributes>"
-    post "/source/home:Iggy:branches:BaseDistro/pack_new/_attribute", params: data
-    assert_response :forbidden
-
-    # when project maintainers are allowed to add Issue attribute
-    attrib_type = AttribType.where(name: "Issues").first
-    attrib_type.attrib_type_modifiable_bies.create(role: Role.where(title: "maintainer").first)
```
Comment (mschnitzer) [inline src/api/test/functional/issue_controller_test.rb]: Why got this test removed?

**Comment by mschnitzer** [comment]
LGTM, but as @evanrolfe mentioned, maybe someone with more knowledge regarding the API should review it. Just looked at the code and checked if it makes sense to me.

File: src/api/test/functional/issue_controller_test.rb (hunk 2 above)
Comment (hennevogel) [inline src/api/test/functional/issue_controller_test.rb]: This was a git revert of the changes you and Björn have introduced based on the 'wrong' attrib policy.

**Comment by hennevogel** [comment]
BTW: All of this is the result of me talking to @adrianschroeter about it :-)

**Comment by adrianschroeter** [comment]
I have to admit that I lost the overview due to the amount of changes here a bit, but yes it seems to restore the old behaviour:

* modifiable_by in attribute definition namespace route (/attribute/$NAMESPACE/_meta) defines only the permission to create, modify or remove attribute *definitions* below.
* modifiable_by in attribute definition route (/attribute/$NAMESPACE/$ATTRIBUTE_NAME/_meta) defines who can create, modify or remove an attribute in the project or package containers.

Having no modifiable_by in attribute definition does mean the project or package maintainers can set them as default. The attribute definitions namespace has no influence here.

As far as I can see, this pull request is giving us this behaviour again.

**Comment by hennevogel** [comment]
If you look only at 2254fe2253acddc4cb67d54aa606b5b64608be6c and ac52bb1560b916eb1b26deb25af8c94a97587b23 it makes more sense I guess :-)

File: src/api/test/functional/issue_controller_test.rb (hunk 2 above)
Comment (mschnitzer) [inline src/api/test/functional/issue_controller_test.rb]: OK

File: src/api/app/policies/attrib_policy.rb (hunk 1 above)
Comment (hennevogel) [inline src/api/app/policies/attrib_policy.rb]: Because? :-)

**Pull request merged**